lib/Rules.js

```diff
@@ -1,6 +1,7 @@
 var assign = require("object-assign");
 var comparators = require("./comparator");
 var path = require("object-path");
+var _ = require("lodash");
```

Comment:
I'd like to keep lodash out of the project to keep this ultra-light for shipping to the client for isomorphic apps.

Maybe do a duck type check for array/object instead `Object.prototype.toString.call( rule[if] ) === '[object Array]'`